upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -57,6 +94,73 @@ func (_ *AttachISO) CheckChanges(a, e, changes *AttachISO) error {
 }
 
 func (_ *AttachISO) RenderVC(t *vsphere.VSphereAPITarget, a, e, changes *AttachISO) error {
-	glog.Info("AttachISO.RenderVC invoked!")
+	startupScript, err := changes.BootstrapScript.ResourceNodeUp(changes.IG)
+	startupStr, err := startupScript.AsString()
+	if err != nil {
+		return fmt.Errorf("error rendering startup script: %v", err)
+	}
+	dir, err := ioutil.TempDir("", *changes.VM)
+	defer os.RemoveAll(dir)
+
+	isoFile := createISO(changes, startupStr, dir)
+	err = t.Cloud.UploadAndAttachISO(changes.VM, isoFile)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func createUserData(startupStr string, dir string) {
+	// Update the startup script to add the extra spaces for
+	// indentation when copied to the user-data file.
+	strArray := strings.Split(startupStr, "\n")
+	for i, str := range strArray {
+		if len(str) > 0 {
+			strArray[i] = "       " + str
+		}
+	}
+	startupStr = strings.Join(strArray, "\n")
+
+	data := strings.Replace(userData, "$SCRIPT", startupStr, -1)
```

Comment:
It would be better to do this using go templates. If you don't want to make this change in this PR, please feel free to add an issue for improving this method and createMetaData  to make use of template.